### PR TITLE
Stats: Remove "checkout-flows-v2" feature flag

### PR DESCRIPTION
**Pull request by a8ck3n**
Related to:

https://github.com/Automattic/red-team/issues/45

## Proposed Changes

Fully removes the `stats/checkout-flows-v2` flag now that the feature has been released.

## Why are these changes being made?

Part of our regular maintenance work to keep tech debt low.

## Testing Instructions

The code changes are pretty straight-forward but it touches a bunch of our controller logic. Please test that the various Stats top-level tabs (Traffic, Insights, etc.) load correctly. 

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by kangzj — APPROVED** [review]
It works well and looks good. Probably get another pair of eyes on it before merging as it's about the checkout flow @Automattic/red

File: client/my-sites/stats/controller.jsx
Comment (grzegorz-cp) [inline client/my-sites/stats/controller.jsx:143]: Why do we need to keep the other part of the condition (lines 156:161)?

Comment (kangzj) [reply client/my-sites/stats/controller.jsx:143]: if no site is specified, it shows an overview of all sites, which is a feature I believe.

image

Comment (grzegorz-cp) [reply client/my-sites/stats/controller.jsx:143]: It points to the same component with the same properties, so I think that the check should have been applied to both cases. This would mean that the second part of the condition should have been removed with the feature flag.

Comment (a8ck3n) [reply client/my-sites/stats/controller.jsx:143]: Thanks for the catch @grzegorz-cp. I'll do some additional testing around this before merging.

Comment (a8ck3n) [reply client/my-sites/stats/controller.jsx:143]: So in my testing, the page doesn't load correctly if run through the `LoadStatsPage` component. I'm seeing this instead of the overview page:

SCR-20240605-qjnr

It looks like some of the tests in the component expect `true` or `false` but instead get set to `null` by the call to `canCurrentUser()`. For example, line 74 results in a `null` value:

```
	const canUserManageOptions = useSelector( ( state ) =>
		canCurrentUser( state, siteId, 'manage_options' )
	);
```

I'll merge this PR as is and then have a closer look at this behaviour for a follow-up PR.

Comment (a8ck3n) [reply client/my-sites/stats/controller.jsx:143]: Created a follow-up Issue to investigate the permission checks and remove the second code path [here](https://github.com/Automattic/red-team/issues/50).

**Pull request merged**
